### ISSUE TITLE
Spyder freezes when quitting help subwindow

### ISSUE THREAD

**Issue by eloilit** [spyder-ide/spyder#7468]
### Issue Report Checklist

* [X] Searched the [issues page](https://github.com/spyder-ide/spyder/issues?q=is%3Aissue) for similar reports
* [X] Read the relevant sections of the [Spyder Troubleshooting Guide](https://github.com/spyder-ide/spyder/wiki/Troubleshooting-Guide-and-FAQ) and followed its advice
* [X] Reproduced the issue after updating with ``conda update spyder`` (or ``pip``, if not using Anaconda)
* [ ] Could not reproduce inside ``jupyter qtconsole`` (if console-related)
* [X] Tried basic troubleshooting (if a bug/error)
    * [X] Restarted Spyder
    * [X] Reset preferences with ``spyder --reset``
    * [X] Reinstalled the latest version of [Anaconda](https://www.anaconda.com/download/)
    * [X] Tried the other applicable steps from the Troubleshooting Guide
* [ ] Completed the **Problem Description**, **Steps to Reproduce** and **Version** sections below

## Problem Description

When closing the "help subwindow" in Spyder, Spyder kind of freezes : you can't click in the IPython Console or in the script subwindow but if you click in the toolbar, you can see the submenus.

Reopening the "help subwindow" through Ctrl+I solves the problem. 

Alternatively, strangely, so does quitting the IPython Console. I mean, "shadow clicking" on the IPython Console closing cross and relaunching Spyder. Indeed, the freeze makes you unable to direct quit the IPython Console but after relaunching, neither the console nor the "help subwindow" are open and it doesn't freeze until you relaunch the Console.

### What steps reproduce the problem?

1. Quitting the "help subwindow"

### What is the expected output? What do you see instead?

No "output", only a good work of Spyder. Instead, Spyder freezes

### Paste Traceback/Error Below (if applicable)

```python-traceback

No traceback

```

## Versions

* Spyder version: 3.2.8
* Python version: 3.6.5 64bits
* Qt version: 5.9.4
* PyQt version: 5.9.2
* Operating System name/version: Windows 10 

### Dependencies

```
IPython >=4.0    :  6.4.0 (OK)
cython >=0.21    :  0.28.2 (OK)
jedi >=0.9.0     :  0.12.0 (OK)
nbconvert >=4.0  :  5.3.1 (OK)
numpy >=1.7      :  1.14.3 (OK)
pandas >=0.13.1  :  0.23.0 (OK)
pycodestyle >=2.3:  2.4.0 (OK)
pyflakes >=0.6.0 :  1.6.0 (OK)
pygments >=2.0   :  2.2.0 (OK)
pylint >=0.25    :  1.8.4 (OK)
qtconsole >=4.2.0:  4.3.1 (OK)
rope >=0.9.4     :  0.10.7 (OK)
sphinx >=0.6.6   :  1.7.4 (OK)
sympy >=0.7.3    :  1.1.1 (OK)

```

**Comment by ccordoba12** [comment]
@CAM-Gerlach, are you able to reproduce this one?

**Comment by CAM-Gerlach** [comment]
@ccordoba12 I tested on both my Win 8.1 and Win 10 (as the OP) machines, with the same versions of Spyder, Python, Qt, PyQt, OS, IPython, jedi, qtconsole, sphinx, etc., and with ``spyder --reset`` for clean prefs and the window layout reset to the default. The testing was done on a standard 1080p screen with no scaling applied. The test consisted of closing the open help pane by the top right "x" close button, and then ensuring the Editor, Console and rest of the UI were functional and responsive. However, I was unable to reproduce this behavior or anything resembling it, nor have I ever experienced it in actual usage.

I tried (all at least once, and a large number of combinations of most of them):

* different functions (``print()`` and ``open()`` with short and long docs respectively), 
* help from both the Editor and the console, 
* on a help page that was loaded, loading or with no documentation found, 
* with the "lock" toggle engaged or disengaged
* with the help pane both in its normal position and "popped out" as a separate window

@eloilit Can you reset the window layout to default, restart, and see if the problem persists? Can you describe precisely, step by step from Spyder launched after ``spyder --reset``, what will reproduce the issue? Anything non-standard in your environment, setup, actions or screen configuration that might trigger this? Thanks.

**Comment by eloilit** [comment]
@ccordoba12 @CAM-Gerlach Thank you for your answers and the time you gave to my poblem.

After reseting the window layout to default and restarting, the problem persists. 
Following your last questions, I tried to understand if my double-screen configuration could play a role.
And it definitively plays a role!

When I disconnect my additional screen, the problem doesn't persist.
More precisely, the following sequence of actions, from Spyder launched after spyder -- reset

1.  Close help pane
2.  Try to click somewhere in Spyder

behaves good with 1-screen configuration while the problem persists with the additional screen connected

So, I tried to mix up 1-screen and 2-screen configurations, doing the following sequence

1. With 1-screen configuration launch spyder after spyder--reset
2. Close the help pane
3. Remark that Spyder works properly
4. Connect the additional screen
5. Remark that Spyder works properly
6. Close Spyder
7. Launch Spyder (no spyder --reset, the help pane remains closed)
8. Remark that Spyder doesn't work properly
9. Disconnect the additional screen
10. Remark that Spyder doesn't work properly
11. Close Spyder
12. Launch Spyder
13. Remark that Spyder works properly

Alternatively, I did the following

1. With 2-screens configuration launch spyder after spyder--reset
2. Close the help pane
3. Remark that Spyder doesn't work properly
4. Disconnect the additional screen
5. Remark that Spyder doesn't work properly
6. Close Spyder
7. Launch Spyder (no spyder --reset, the help pane remains closed)
8. Remark that Spyder  works properly
9. Connect the additional screen
10. Remark that Spyder  works properly
11. Close Spyder
12. Launch Spyder
13. Remark that Spyder doesn't work properly

I wish it will help you understand the problem ;)

eloilit

**Comment by CAM-Gerlach** [comment]
Thanks for your information. To summarize the above, it seems the determining factor as to whether the problem presents itself in your setup is whether Spyder was launched with the second screen attached, regardless of whether the same is still attached at the time of the test procedure.

As I have used Spyder for countless hours under Windows 10 with two-and even three-screen setups with both homogeneous and heterogeneous screens and never seen this issue (as have our many other users), it seems quite possible it is something specific to your setup, and quite possibly related to #7197 or maybe #7391 , considering they only manifested when multiple monitors where attached and closing the help pane was the key element  (in that case, *fixed*) those problems.

Are either of your monitors High-DPI (> 1200p)? Are either of them scaled? Did you set any scaling settings in Windows? What are their resolutions? What GPU are you using? Thanks.

**Comment by eloilit** [comment]
![setup_1](https://user-images.githubusercontent.com/34735700/43190328-0e4ada72-8ff9-11e8-8d04-a7724db3e5fe.PNG)
![setup_2](https://user-images.githubusercontent.com/34735700/43190327-0e33541a-8ff9-11e8-8bfd-b2a7db12855e.PNG)
![setup_3](https://user-images.githubusercontent.com/34735700/43190326-0e1c4036-8ff9-11e8-9b2a-6eada769f51d.PNG)
![setup_4](https://user-images.githubusercontent.com/34735700/43190329-0e66dc72-8ff9-11e8-8da6-a74cf88d22df.PNG)

Here are my whole setup informations!

**Comment by CAM-Gerlach** [comment]
Thanks for all the information. That all looks very standard, except for one monitor being driven off the Intel iGPU and one off the Nvidia (and correspondingly different driver stacks, etc). Assuming you can (it looks like they are both VGA for some reason), can you connect both to one or the other to make sure that doesn't have anything to do with it?

**Comment by eloilit** [comment]
The only thing I have to add is that I use a HDMI-VGA AUKEY converter. Indeed, the DELL screen has only a VGA port.

**Comment by CAM-Gerlach** [comment]
Hmm. Is the first monitor built in, so you can't switch it? Also, I looked up the P2414H, and it says it has both DVI-D and DisplayPort (1.2a) along with the VGA you mentioned, and the reference 1050 Ti has both those ports (Dual-link DVI and DisplayPort 1.4) along with HDMI so you should be fine. 

Somewhat OT, but I'm utterly baffled why one would use a converter (and an expensive and potentially problematic *active* converter, no less) to convert to an archaic analog standard that's been superseded for the past 20 years by three successively superior digital standards (DVI, HDMI and DP), when both the card and monitor involved should offer at least two of those latter ports in common, and the cables are much cheaper, much more common and much more likely to be included with the monitor (the information I have says they should have been with a new P2414H) than an active adapter. Is there something very strange going on with your setup that might somehow require this?

**Comment by eloilit** [comment]
Yes it is. It switches automatically when it needs to the NVIDIA. I will try to switch the additional screen to the built-in Intel graphics.

The "strange" thing is that I'm currently in internship in a company where they all have DELL PC linked to the screen with DELL stations, while I use my own computer that only has HDMI, USB-C and USB ports. The company only found VGA-VGA cables and I then bought the VGA-HDMI converter...

**Comment by CAM-Gerlach** [comment]
Oh okay, so you have a laptop then with a mobile 1050 Ti? It didn't clearly indicate that in the dialog; the Nvidia 1000-series has been really bad about that for some reason.

Try updating Spyder and Qt/PyQt? ``conda update spyder pyqt qt`` from the Anaconda prompt in the appropriate environment.

**Comment by eloilit** [comment]
Yup
